From c6c99ace8addc4d68d098fb5b03b7a36dd9f2eae Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Thu, 28 Nov 2024 11:03:06 +0100
Subject: [PATCH 750/954] fix(stm32mp2): fix shutdown on STM32MP21

Simplify code for STM32MP21 in stm32_system_off, as only one core
is supported.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 672cf26c6d2cb81994ca01df7f5b2e5b76ba3b63 fix(stm32mp2): fix shutdown on STM32MP21]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: Iff4512491d380f3fc7916ed184bcd41254f71a2c
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/421336
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 plat/st/stm32mp2/stm32mp2_pm.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/plat/st/stm32mp2/stm32mp2_pm.c b/plat/st/stm32mp2/stm32mp2_pm.c
index c975445bab..893f733cd0 100644
--- a/plat/st/stm32mp2/stm32mp2_pm.c
+++ b/plat/st/stm32mp2/stm32mp2_pm.c
@@ -902,6 +902,7 @@ static void __dead2 stm32_system_off(void)
 		panic();
 	}
 
+#if !STM32MP21
 	if (!stm32_freeze_other_core(core_id)) {
 		WARN("PSCI system off with other core running.\n");
 
@@ -915,14 +916,13 @@ static void __dead2 stm32_system_off(void)
 		/* After reset, the core is stopped, waiting in WFI loop */
 		stm32mp_state_set(STM32MP_SECONDARY_CPU, STATE_START, false);
 
-#if !STM32MP21
 		/* Reset the secondary core */
 		mmio_write_32(RCC_BASE + RCC_C1P1RSTCSETR, RCC_C1P1RSTCSETR_C1P1PORRST);
-#endif /* !STM32MP21 */
 
 		/* Forbid access to DDR */
 		stm32mp_state_set(STM32MP_PRIMARY_CPU, STATE_DDR, false);
 	}
+#endif /* !STM32MP21 */
 
 	/* If CPU2 is not in reset */
 	if (stm32_pwr_cpu2_state_is_running(pwr_base)) {
-- 
2.53.0

